### PR TITLE
#18194  fix NPE on site_search_audit.jsp

### PR DESCRIPTION
**Pull request by fabrizzio-dotCMS**
This comes to solve a problem that breaks the jsp and prevents it from loading when site search job option all sites is selected

### CODE REVIEW

File: dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp [hunk 1]
```diff
@@ -57,11 +56,19 @@
...
-            if(UtilMethods.isSet(hid))
-                hostList.append(APILocator.getHostAPI().find(hid, APILocator.getUserAPI().getSystemUser(), false).getHostname())
-                        .append("  ");
+        final StringBuilder hostList = new StringBuilder();
+        final String [] sites = a.getHostList().split(",");
+        if(UtilMethods.isSet(sites) ){
+            if ("empty".equalsIgnoreCase(sites[0])) {
+                hostList.append(LanguageUtil.get(pageContext, "All")).append("  ");
+            } else {
+                for (String hid : sites) {
```
Comment (jdotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:65]: rename to hostId

Comment (jdotcms) [reply dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:65]: set to final

File: dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp [hunk 2]
```diff
@@ -57,11 +56,19 @@
...
     <% for(SiteSearchAudit a : recents) { %>
     <%
-        StringBuilder hostList=new StringBuilder();
-        for(String hid : a.getHostList().split(",")) {
-            if(UtilMethods.isSet(hid))
-                hostList.append(APILocator.getHostAPI().find(hid, APILocator.getUserAPI().getSystemUser(), false).getHostname())
-                        .append("  ");
+        final StringBuilder hostList = new StringBuilder();
+        final String [] sites = a.getHostList().split(",");
+        if(UtilMethods.isSet(sites) ){
```
Comment (jdotcms) [inline dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:61]: sites must check if size is >= 1

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp (hunk 2 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:61]: UtilMethods.isSet(String) does that internally

File: dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp (hunk 1 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:65]: done

File: dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp [hunk 3]
```diff
@@ -57,11 +56,19 @@
     </tr>
     <% for(SiteSearchAudit a : recents) { %>
     <%
-        StringBuilder hostList=new StringBuilder();
-        for(String hid : a.getHostList().split(",")) {
-            if(UtilMethods.isSet(hid))
-                hostList.append(APILocator.getHostAPI().find(hid, APILocator.getUserAPI().getSystemUser(), false).getHostname())
-                        .append("  ");
+        final StringBuilder hostList = new StringBuilder();
+        final String [] sites = a.getHostList().split(",");
```
Comment (freddyucv) [inline dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:60]: what is it happen if getHostList return null?

File: dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp [hunk 4]
```diff
@@ -57,11 +56,19 @@
     </tr>
     <% for(SiteSearchAudit a : recents) { %>
     <%
-        StringBuilder hostList=new StringBuilder();
-        for(String hid : a.getHostList().split(",")) {
-            if(UtilMethods.isSet(hid))
-                hostList.append(APILocator.getHostAPI().find(hid, APILocator.getUserAPI().getSystemUser(), false).getHostname())
-                        .append("  ");
+        final StringBuilder hostList = new StringBuilder();
```
Comment (freddyucv) [inline dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:59]: where is hostList use?

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp (hunk 3 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:60]: shouldn't return null. The Audit data table demands some is written into that field at all times.

File: dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp (hunk 4 above)
Comment (fabrizzio-dotCMS) [reply dotCMS/src/main/webapp/html/portlet/ext/sitesearch/site_search_audit.jsp:59]: hostList is used downstream in the jsp to print the host names

**Pull request merged**